pkg/ccl/serverccl/settings_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package serverccl
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+)
+
+func TestConsoleKeysVisibility(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	for _, key := range settings.ConsoleKeys() {
+		setting, found := settings.LookupForLocalAccessByKey(key, true /* forSystemTenant */)
+		if !found {
+			t.Fatalf("not found: %q", key)
+		}
+
+		if setting.Class() == settings.SystemOnly {
+			t.Errorf("setting %q used in console, cannot be system-only", key)
+		}
+	}
```

Comment:
:D When I opened this I thought, "oh, we have a slice of these, we can test for this" and then I see you beat me to it.